FCCSW_ecal/run_thetamodulemerged.py

```diff
@@ -491,14 +492,34 @@
     createHcalBarrelPositionedCells2.positionedHits.Path = hcalBarrelPositionedCellsName2
 
     # Create cells in HCal endcap
-    # createHcalEndcapCells = CreateCaloCells("CreateHcalEndcapCaloCells",
-    #                                    doCellCalibration=True,
-    #                                    calibTool=calibHcalEndcap,
-    #                                    addCellNoise=False,
-    #                                    filterCellNoise=False,
-    #                                    OutputLevel=INFO)
-    # createHcalEndcapCells.hits.Path="HCalEndcapHits"
-    # createHcalEndcapCells.cells.Path="HCalEndcapCells"
+    # 1 - merge hits into cells with the default readout
+    hcalEndcapCellsName = "HCalEndcapCells"
+    createHcalEndcapCells = CreateCaloCells("CreateHCalEndcapCells",
+                                            doCellCalibration=True,
+                                            calibTool=calibHcalEndcap,
+                                            addCellNoise=False,
+                                            filterCellNoise=False,
+                                            addPosition=True,
+                                            hits="HCalEndcapHits",
+                                            cells=hcalEndcapCellsName,
+                                            OutputLevel=INFO)
+
+    # 2 - attach positions to the cells (cell positions needed for RedoSegmentation!)
```

Comment:
Have you tested that the clustering works on the HCal endcap cells? For the barrel I remember that I had to implement the extra step of merging the cells with different rows but same theta together with the RedoSegmentation because otherwise only one cell would be picked up among the various cells which shared the same theta ID

Reply:
Right, I haven't tested the clustering yet. I wanted to focus on the cells first and get some minimal working version of endcap and the clustering is the next step.